### ISSUE TITLE
Mac OSX - "open with" context menu has all (previous) versions of Atom installed

### ISSUE THREAD

**Issue by Patryk1098** [atom/atom#9774]
Right-clicking on a file within Finder and selecting "Open With..." shows a sub-menu of suggested applications. All previously installed versions of Atom are listed. Only 1 copy of the app lives in Applications, so it's likely each version of Atom is added to the registered applications list when upgraded, but the old version isn't removed.

![atom bug - open with multiple versions](https://cloud.githubusercontent.com/assets/2947116/11353714/7ac09ea8-91fa-11e5-914b-0003c1a68746.png)

**Comment by Patryk1098** [comment]
Oddly, the version numbers are inconsistent. I've done upgrades before, but they didn't seem to take until I selected "restart and install updates" from the apple menu. Confusing because the Squirrel icon (bottom right) always led me to believe that's what I was doing there. Apparently I was still running an older version. After selecting "Restart & Install Updates" from the Apple menu (in Atom), I only saw 1 version in the "Open With" menu.

**Comment by woninja** [comment]
Hey i am also facing the issue. What did you do to get it solved ?

**Comment by Patryk1098** [comment]
@woninja I uninstalled it and re-installed the latest version. Now, the only time I see this issue is when there's an update available. ...At least this is accurate to my recollection. Did you read the whole comment? I mentioned that I did resolve the issue.

**Comment by woninja** [comment]
/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user;killall Finder;echo "Rebuilt Open With, relaunching Finder"

**Comment by jayaprakash-altimetrik** [comment]
Thanks @woninja. Your solution worked perfectly.

**Comment by hongkongkiwi** [comment]
Thanks. It would be great if this command could be automatically run upon an Atom upgrade.... (hint hint)

**Comment by 24Degrees** [comment]
Thanks @woninja - your little script solved a very annoying situation here.

**Comment by patrickcameron** [comment]
For any other newbs here -- copy/paste woninja's entire message into the Terminal and hit enter. Should take a few seconds to run.

**Comment by zefrench** [comment]
Thanks @woninja. Your solution worked perfectly.

**Comment by blakeperdue** [comment]
A better solution is documented here: https://github.com/atom/atom/issues/1720

Run this in terminal: `rm -rf ~/Library/Caches/com.github.atom.ShipIt`

It will only remove the Atom duplicates and won't force Finder to reset all other data it is on your other apps.

**Comment by miladhub** [comment]
This didn't solve the problem for me: `rm -rf ~/Library/Caches/com.github.atom.ShipIt`
This did: `/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user;killall Finder;echo "Rebuilt Open With, relaunching Finder"`

**Comment by CodyDunlap** [comment]
I was having this issue, with 4 previous versions all showing. I manually updated Atom by going to Atom > About > Restart and Install for version 1.19.7 and it seems to have fixed this issue now.

**Comment by DrJekel** [comment]
Hello! I was having the same problem but with VLC media player.

> /System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user;killall Finder;echo "Rebuilt Open With, relaunching Finder"

The above quoted code, written into terminal solved my problem, but after a restart it returned. I also noticed when I tried to open with a previous version, it would actually open, meaning the application was actually still installed somewhere, just not in my applications folder. SO I went hunting for it, I looked in my User/myUserName/Library/Application Support/ and found a VLC folder which contained the old version. Once deleted and trash emptied, THEN the above code worked and persisted through a restart. Perhaps this methodology would apply to other programs as well. Good luck to anyone else finding this issue.
